client/assets/calendar.js

```diff
@@ -0,0 +1,121 @@
+		right_now += ((moment().month() + 1).toString().length == 2) ? moment().month() + 1 : '0' + (moment().month() + 1)
+		right_now += (moment().date().toString().length == 2) ? moment().date() : '0' + moment().date()
+		right_now += 'T'
+		right_now += (moment().hours().toString().length == 2) ? moment().hours() : '0' + moment().hours()
+		right_now += (moment().minutes().toString().length == 2) ? moment().minutes() : '0' + moment().minutes()
+		right_now += '00Z'
+
+		//set up timezone info for this event
+		calendar_string += 'BEGIN:VEVENT\nDTSTART;TZID=America/Los_Angeles:'
+		calendar_string += start_moment.year()
+		calendar_string += ((start_moment.month() + 1).toString().length == 2) ? start_moment.month() + 1 : '0' + (start_moment.month() + 1)
+		calendar_string += (start_moment.date().toString().length == 2) ? start_moment.date() : '0' + start_moment.date()
+		calendar_string += 'T'
+		calendar_string += (start_moment.hours().toString().length == 2) ? start_moment.hours() : '0' + start_moment.hours()
+		calendar_string += (start_moment.minutes().toString().length == 2) ? start_moment.minutes() : '0' + start_moment.minutes()
+		calendar_string += '00'
+		calendar_string += '\nDTEND;TZID=America/Los_Angeles:'
+		calendar_string += start_moment.year()
+		calendar_string += ((start_moment.month() + 1).toString().length == 2) ? start_moment.month() + 1 : '0' + (start_moment.month() + 1)
+		calendar_string += (start_moment.date().toString().length == 2) ? start_moment.date() : '0' + start_moment.date()
+		calendar_string += 'T'
+		calendar_string += (end_moment.hours().toString().length == 2) ? end_moment.hours() : '0' + end_moment.hours()
+		calendar_string += (end_moment.minutes().toString().length == 2) ? end_moment.minutes() : '0' + end_moment.minutes()
+		calendar_string += '00'
+		//set what days this course is on, aka the repititionss
+		calendar_string += '\nRRULE:FREQ=WEEKLY;WKST=SU;UNTIL='
+		calendar_string += end_moment.year() 
+		calendar_string += ((end_moment.month() + 1).toString().length == 2) ? end_moment.month() + 1 : '0' + (end_moment.month() + 1)
+		calendar_string += (end_moment.date().toString().length == 2) ? end_moment.date() : '0' + end_moment.date()
+		calendar_string += 'T105959Z;BYDAY='
+		course_dows.forEach(function(dow) {
+			calendar_string += ics_day_list[dow] + ','
+		})
+		calendar_string = calendar_string.slice(0, -1) //remove the extra comma at the end
+		//other event information
+		calendar_string += '\nDTSTAMP:' + right_now + '\nUID:' + plan_course.title + '@usf.nickthesick.com\nCREATED:' + right_now
+		calendar_string += '\nLAST-MODIFIED:' + right_now + '\nLOCATION:' + plan_course.loc + '\nSEQUENCE:0\nSTATUS:CONFIRMED\nSUMMARY:'
+		calendar_string += plan_course.title + '\nEND:VEVENT\n'
+	})
+	calendar_string += 'END:VCALENDAR\n'
+
+	var blob = new Blob([calendar_string], {type: "text/plain;charset=utf-8"});
+	FileSaver.saveAs(blob, plan.title + ".ics");
```

Comment:
Does this ever fail at all? If it does this function should swallow the error and return false if it failed and true if it succeeded so I can tell the user later if it was successful or not

Reply:
the only way it "fails" is by making an unreadable ICS file, which will only become known when the user tries to open it. Regardless, I surrounded it with a try catch